detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/Project/ProjectXmlResolver.cs

```diff
@@ -102,6 +102,11 @@ public DependencyResult Process()
                 }
             }
 
+            if (packagesNodes.Count > 0 && result.Packages.Count == 0)
+            {
```

Comment:
The changes to the files ProjectXmlResolver.cs and ProjectReferenceResolver.cs are handling the last testcase. Whenever these references are able to find some packages via parsing but are not able to resolve them are also considered a BadParse since there are package which resolvers are not able to find.